### DISCUSSION TITLE
Editor: Beat divisor visual representation

### DISCUSSION THREAD

**Discussion by Gyoshi** [ppy/osu#23527] [Ideas] [answered]
I feel like the beat divisor should be visually represented in the same way as they are seen on the timeline, with the higher divisors getting closer and closer to the white beat line.

Current:
![image](https://github.com/ppy/osu/assets/41743877/c87d808b-6cea-4915-87eb-abe53538022c)
Better:
![better divisor mockup](https://github.com/ppy/osu/assets/41743877/492de0b1-8287-4efb-b04c-ab3da0d74a90)

P.S. In my paint mockup I added a white line on the right as a visual reference point (which I think is necessary in this kind of representation), but it should probably be symmetric with the left line unlike how I drew it. [Edit: Although since only the left white line should be a selectable option, perhaps having the second white line be partially cut off does make sense.]

[Edit2: the fact that the selection loops around actually works pretty well with displaying an extra white line]

**Comment by bdach** [comment]
[The designs](https://www.figma.com/file/ytnnne2TH8Z956Jxhiypqq/Beatmap-Editor-2?type=design&node-id=1-18&t=GRW6o0O3jkYKflHf-4) had it a similar way, I think we've just never bothered to update the crude implementation from however long which is still in place.

Mind you, I'd probably do your "better" version but horizontally flipped. As in, 1/8 should be 1/8 of the width between the white bars from the left, rather than 7/8ths.

**Reply by Gyoshi** [reply]
I agree having 1/8 over 7/8 makes sense.

But finer divisors being to the right matches stable. Also I think it is intuitive for more complex (more extreme) values to be to the right.

**Comment by peppy** [comment, marked as answer]
#23581
